src/checksums.ts

```diff
@@ -6,9 +6,165 @@ const httpc = new httpm.HttpClient(
   {allowRetries: true, maxRetries: 3}
 )
 
+/** Known checksums from previously published Wrapper versions */
+export const KNOWN_VALID_CHECKSUMS = new Set([
```

Comment:
Is there a way to pull this from a resource file instead of hard-coding it into the source code.

Ideally, there would be a resource file, and there would also be a script that CI could automatically run to update this file and would auto-push then auto-release when there was an update

Reply:
Yep, that would be much better. If the file was formatted as JSON, it could be trivially loaded into an object: that way we could include the Gradle version as well as the checksum in the metadata.

@Marcono1234 you can [see here](https://github.com/gradle/actions/blob/main/sources/src/cache-base.ts#L236-L241) an example of loading a resource file in Typescript. It may not be the best way, but it works!
Then, you can use `JSON.parse(fileContents)` and use it as a typed Javascript object. Here [are](https://github.com/gradle/actions/blob/aff52e5be96935327d77c5529075184377dc4371/sources/src/build-results.ts#L25) [some](https://github.com/gradle/actions/blob/aff52e5be96935327d77c5529075184377dc4371/sources/src/provision.ts#L168) [examples](https://github.com/gradle/actions/blob/aff52e5be96935327d77c5529075184377dc4371/sources/src/cache-extract-entries.ts#L281).